flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/SpecificInputTypeStrategies.java

```diff
@@ -71,7 +71,11 @@ public final class SpecificInputTypeStrategies {
 
     /** Argument type derived from the array element type. */
     public static final ArgumentTypeStrategy ARRAY_ELEMENT_ARG =
-            new ArrayElementArgumentTypeStrategy();
+            new ArrayElementArgumentTypeStrategy(false);
+
+    /** Argument type derived from the array element type. But leaves nullability untouched. */
```

Comment:
that seems not true.
Example: array[1, 2, 3] which has nullability `NOT NULL` and `ARRAY_ELEMENT_ARG_NULLABLE` makes it nullable. So it looks like it doesn't "untouched" nullability.

Reply:
Compare implementation. The logic in implementation is different: in `org.apache.flink.table.types.inference.strategies.CommonArgumentTypeStrategy#inferArgumentType` it could preserve nullability. In your case no, that's why i raised it

Reply:
the reason why i add ARRAY_ELEMENT_ARG and ARRAY_ELEMENT_ARG_NULLABLE like COMMON_ARG/COMMON_ARG_NULLABLE is some  array function may need element arg not null. so i preserve.

Reply:
thanks for your explain in detail. i will delete this comment "But leaves nullability untouched"

Reply:
i guess this change could be completely removed once the issue with `ArrayElementArgumentTypeStrategy#inferArgumentType` is solved

Reply:
yeap.